sdk/containerregistry/Azure.Containers.ContainerRegistry/tests/ContainerRepositoryClientLiveTests.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading.Tasks;
+using Azure.Core.TestFramework;
+using NUnit.Framework;
+
+namespace Azure.Containers.ContainerRegistry.Tests
+{
+    public class ContainerRepositoryClientLiveTests : RecordedTestBase<ContainerRegistryTestEnvironment>
+    {
+        private readonly string _repositoryName = "library/hello-world";
+
+        public ContainerRepositoryClientLiveTests(bool isAsync) : base(isAsync)
+        {
+        }
+
+        private ContainerRepositoryClient CreateClient()
+        {
+            return InstrumentClient(new ContainerRepositoryClient(
+                new Uri(TestEnvironment.Endpoint),
+                _repositoryName,
+                TestEnvironment.UserName,
+                TestEnvironment.Password,
+                InstrumentClientOptions(new ContainerRegistryClientOptions())
+            ));
+        }
+
+        [RecordedTest]
+        public async Task CanGetRepositoryProperties()
+        {
+            var client = CreateClient();
+
+            RepositoryProperties properties = await client.GetPropertiesAsync();
+
+            Assert.AreEqual(_repositoryName, properties.Name);
+            Assert.AreEqual(new Uri(TestEnvironment.Endpoint).Host, properties.Registry);
+        }
+
+        [RecordedTest]
+        public async Task CanSetRepositoryProperties()
```

Comment:
Just a random tip that I learned recently: 
If you wanted to test all permutations of the properties being set, you could do something [like this](https://github.com/Azure/azure-sdk-for-net/blob/de78ef39a9c5e21ad87fd34147f599425f7ff34e/sdk/identity/Azure.Identity/tests/DefaultAzureCredentialTests.cs#L329-L336)

Then you could call `SetProperties` with the values of the test arguments and assert that they are the same after `GetProperties`

Reply:
Very cool, thank you!